src/routes/register/index.tsx

```diff
@@ -75,10 +84,14 @@ function Register({ register }: RegisterProps) {
           Submit
         </Button>
       </form>
+      <div className={styles.errorContainer}>
+        {errors.map((text: string, index) => ErrorSnack(text, index))}
```

Comment:
This is not quite idiomatic React, although, as you can see, it does work. Instead, you would want to make `ErrorSnack` a component and pass `text` and `index` as props:

```js
// Props are passed into ErrorSnack and we destructure the props
function ErrorSnack({ text, index }) {
  return (
    <div className={styles.error} key={`text-${index}`}>
      {text}
    </div>
  );
}

// Then when mapping over the errors...
{errors.map((text: string, index) => <ErrorSnack text={text} index={index} /> )}
```

Also, you could actually just use the `text` as the key and then not have to worry about passing in the index. The reason you never really want to use an index is because React uses keys to track changes to components in a list (this is one of the ways that React is so fast at diffing changes). If an array changes in size, the index of its elements can change and that can lead to weird bugs.

Reply:
Got it, thanks for the explanation. I've went ahead and made it a component.

Reply:
On a side note, the reason I concatted index with text was to not have two matching keys (i.e. when the same error is thrown twice). Regardless, I've went ahead and killed index.